source/js/queue.js

```diff
@@ -0,0 +1,45 @@
+function Queue() {
+    this.dataStore = [];
+    this.offer = offer;
+    this.poll = poll;
+    this.execNext = execNext;
+    this.debug = false;
+    this.startDebug = startDebug;
+
+
+    function offer(element) {
+        if(this.debug) console.log("Offered a Queued Function.");
```

Comment:
- Please user single quote inside JavaScript code (change it for entire file)
- Please surround `if` statement with brackets even for one line, and display it in several lines (change it for entire file)